mautic/mautic
# Phpstan

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

It makes sense that after this great refactoring we should use tools that will help us discover issues. PHPSTAN is perfect for that. We couldn't use it on Mautic 2 as it's PHP 7+ only but we can use it now. This PR adds it to the dev dependencies, runs it on Travis CI. It required some changes. I did the simple changes in this PR and I created other issues for those that will take some time.

I also broke running PHPSTAN and CS Fixer into separate Travis CI jobs. Check how Travis looks like now.

This PR must be rebased once those issues are solved:
- [x] https://github.com/mautic/mautic/pull/8270
- [x] https://github.com/mautic/mautic/issues/8297
- [x] https://github.com/mautic/mautic/issues/8294
- [x] https://github.com/mautic/mautic/issues/8293
- [x] https://github.com/mautic/mautic/issues/8292

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Make sure Mautic loads. Click around. No error should appear.
